pkg/test/framework/components/namespace/namespace.go

```diff
@@ -22,6 +22,29 @@ import (
 	"istio.io/istio/pkg/test/framework/resource"
 )
 
+// Config contains configuration information about the namespace instance
+// including naming information, sidecar injection config, and other component-specific labels
+// note: each component that adds additional custom k8s labels for control should have a field here
+type Config struct {
+	Prefix        string
+	SidecarConfig SidecarConfig
+	CitadelConfig CitadelConfig
+}
+
+// CitadelConfig contains the values for configuration of citadel-specific namespace labels
+// if not included in config, namespace will have none of these labels
+type CitadelConfig struct {
```

Comment:
There should be nothing named `Citadel` in the namespace package ... namespaces don't know about other components.

Reply:
Are you suggesting that we simply rename the config options here not to include `Citadel` in the name (i.e. `CaManaged` as top-level field rather than having `Managed` field inside `CitadelConfig` struct), or that we get these configs out of the `namespace` package entirely? I'm not sure I see a big difference between `InjectionConfig` and `CaConfig` here, or why the former would make sense to include but not the latter.

Because we directly control other components' behavior through namespace labels, the namespace interface we provide in tests might need some awareness of the other components, no?

Reply:
If we need to support arbitrary labels, perhaps we should just have a map of labels on the config object and call it a day?

Reply:
I'm a fan of having this represented as separate fields in config and then having a [translation func](https://github.com/istio/istio/pull/16090/files#diff-e18d43273cb5685dcb060a8396ac9b5dR116) into map for a few reasons: 

1. Provide explicit reference of which labels are togglable from the test framework (the whole label/annotation API is already too implicit/undocumented)
2. Single-file change if the label we use for some config changes
3. No idea how these config knobs will be turned in non-k8s environment (or if it will even be related to namespace, actually)

Thoughts? I'm also fine with just doing a map if this is too involved.

Reply:
I'm not against having explicit fields in the config object so long as they're not specific to another component

If we need component-specific labels there are essentially 2 options:

1. all labels are just provided as a map
2. a mix: some generic fields that get converted to labels and then an additional map for component-specific labels.